### PR TITLE
Add non-blocking vibration to notifications via MotorController component

### PR DESCRIPTION
**Pull request by jlukanc1**
Adds a component to simply control the motor using 2 commands:
Init() sets up the timers
setDuration(duration) triggers the timer for the specified duration

These are also added to the notifications app.

Improves the piece from #155 
Fixes the problems from [JF's requests](https://github.com/JF002/Pinetime/pull/155#issuecomment-759726329) (hopefully!)

### CODE REVIEW

**Comment by petterhs** [comment]
Nice! This works for me. will close #155 now

**Comment by JF002** [comment]
Thanks for this PR. I've already added a few comments. I mostly looks good! 
The only thing that worries me is the request to lfclk.

**Comment by jlukanc1** [comment]
The lfclk request is since the softdevice is not used, you have to manually request it, according to the nRF docs anyways.

Im not sure if it could be requested somewhere else or not.

**Comment by JF002** [comment]
It's only needed for the timer app? 
As we must request it, shouldn't we "unrequest" it when it's not needed anymore?

**Comment by jlukanc1** [comment]
I'm not sure, looking at the docs, it seems that it could cancel during another timer, and when the timer is restarted, it may restart interrupted ones?

**Comment by JF002** [comment]
I need to read the reference manual again to remember how clocks work :)

**Comment by jlukanc1** [comment]
Oh! It looks like it automatically stops when there are no more requests.

File: src/components/motor/MotorController.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "app_timer.h"
+#include "nrf_drv_clock.h"
+
+APP_TIMER_DEF(vibTimer);
+
+using namespace Pinetime::Controllers;
+
+static void lfclk_request(void)     //get the low freq. clock
+{
+    nrf_drv_clock_init();
```
[the remark this answers is not in the document]
Comment (jlukanc1) [reply src/components/motor/MotorController.cpp:13]: It looks like the clock stops when there are no app timers.

File: src/components/motor/MotorController.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#include "MotorController.h"
+#include <hal/nrf_gpio.h>
+#include "systemtask/SystemTask.h"
+#include "app_timer.h"
+#include "nrf_drv_clock.h"
+
+APP_TIMER_DEF(vibTimer);
```
[the remark this answers is not in the document]
Comment (jlukanc1) [reply src/components/motor/MotorController.cpp:7]: Should it be public or private?

File: src/displayapp/screens/Notifications.h [hunk 3]
```diff
@@ -45,6 +46,7 @@ namespace Pinetime {
             const char* text;
           };
           Pinetime::Controllers::NotificationManager& notificationManager;
+          Pinetime::Controllers::MotorController motorController;
```
[the remark this answers is not in the document]
Comment (jlukanc1) [reply src/displayapp/screens/Notifications.h:49]: I'm not too sure how to do this correctly. Sorry about that.

**Comment by petterhs** [comment]
I don't know what to do about APP_TIMER_DEF(vibTimer); and if we need to do anything about it.
It works now and the motorcontroller can be passed to the applications wanting vibrations.

**Comment by JF002** [comment]
As I said in the chat : this macro (APP_TIMER_DEF) is a macro of macros and does more than just instanciating the variable so it won't fit in the header of the class.
I do not see any c++ way to use this macro, so you can just leave it in the CPP file. Maybe we'll refactor this later on (a solution would be to not use the NRF SDK and re-implement the timer driver natively in C++).

I still have to have a look at these clock function. I would like to check what they do to ensure they won't create weird bugs or prevent sleep mode.

**Comment by petterhs** [comment]
I have been using this PR for a while now and have some suggestions:

- JF was right. The clock functions are not needed. I removed them and it still works. 

- As of now the trigger of the vibration happens in the Notification screen. This means that if the notification screen does not pop up, the vibration will not trigger. This is the case if you are currenty in another screen than Clock or if the watch went into sleep mode from another screen than Clock. If we instead trigger the vibration in the SystemTask then we don't need to pass Motorcontroller to Display app and Notification screen and we will get notifications as they are received not shown.

I have done this in this branch: https://github.com/petterhs/Pinetime/tree/theking-upstream-dev
(Also refactored the motorcontroller class a bit)

Created this so you can merge @jlukanc1 if you like the changes: https://github.com/jlukanc1/Pinetime-infini/pull/1

**Comment by jlukanc1** [comment]
@petterhs They are merged!

**Comment by petterhs** [comment]
Added a temporary commit to be able to toggle the vibration on and off. When the watch is charging the vibration can be a bit annoying and I figure it can be smart to have a way of turning it of. As i understand, long term we want a config file using littlefs that we can change settings in, so this is a temporary solution.

Vibrations are toggled by long pressing notification screen. Either the notification preview or from the notification app. Used the NotificationManager class to do this as it is shared in the Notification screen and in Systemtask where vibrations are triggered.

Wanted to add a muted icon on the notification screen to show if vibrations were off but didn't find an easy way of doing this.

Commit is here: https://github.com/petterhs/Pinetime/tree/theking-upstream-dev if it is wanted

**Comment by jlukanc1** [comment]
@petterhs Also merged! 👍

**Comment by JF002** [comment]
@jlukanc1  @petterhs thank you very much for your good work and your patience!

**Pull request merged**